pkg/ui/handlers.go

```diff
@@ -1,23 +1,32 @@
 package ui
 
 import (
+	"fmt"
 	"io/fs"
 	"net/http"
 	"path"
+	"regexp"
+	"strings"
 
 	"github.com/canonical/identity-platform-login-ui/internal/logging"
 	"github.com/go-chi/chi/v5"
 )
 
 type API struct {
 	fileServer http.Handler
-
-	logger logging.LoggerInterface
+	fileSystem fs.FS
+	logger     logging.LoggerInterface
 }
 
 func (a *API) RegisterEndpoints(mux *chi.Mux) {
 	// TODO @shipperizer unsure if we deal with any POST/PUT/PATCH via js
-	mux.HandleFunc("/*", a.uiFiles)
+	nodeList, _ := fs.ReadDir(a.fileSystem, ".")
+	r, _ := regexp.Compile(`html$`)
```

Comment:
if u got to compile it and reuse it (even though in practice we call it only once, but potentially we could multiple times), better to put it in the `constructor NewAPI`

Reply:
I compiled the regex because we use it in a loop, so we match it with every file in ui/dist.